### DISCUSSION TITLE
Conditional matching: more than one way to "replace" based on condition.

### DISCUSSION THREAD

**Discussion by rflcrz** [espanso/espanso#1202] [Feature requests and Ideas]
Hi, @federico-terzi.

What about conditional matching? Something similar to PhraseExpress [Conditional statement](https://www.phraseexpress.com/mac/doc/macros/programming/#conditional-statement).

A "condition" could be added under "replace". In this case, an expression could be evaluated as a shell extension. The first "replace" with a valid "condition" would be used.

We would be able to do OS-specific matches _and more_!

Additionally to a shell expression, there could be expressions to evaluate user defined {{variables}} that would give us, for example, chance to make gender and number agreement on a sentence.

Thank you for this great project!

_Originally posted by @rflcrz in https://github.com/federico-terzi/espanso/issues/276#issuecomment-752772827_

**Comment by federico-terzi** [comment]
@rflcrz Thank you for the feedback! I'll add it to the feature list :)

**Comment by SpecialCharacter** [comment]
I highy support this idea! :)
